eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/processors/SubSnpNoHgvsToDbsnpVariantsWrapperProcessorTest.java

```diff
@@ -60,31 +81,87 @@ public static void tearDownClass() throws Exception {
 
     @Test
     public void processSubmittedVariantAllelesMismatch() throws Exception {
-        SubSnpNoHgvs subSnpNoHgvs = new SubSnpNoHgvs(25928972L, 14718243L, "A", "C", ASSEMBLY, BATCH_HANDLE,
-                                                     BATCH_NAME, CHROMOSOME, CHROMOSOME_START, CONTIG_NAME,
-                                                     CONTIG_START, DbsnpVariantType.SNV, Orientation.FORWARD,
-                                                     Orientation.FORWARD, Orientation.FORWARD, false, false, false,
-                                                     false, CREATED_DATE, CREATED_DATE, TAXONOMY);
+        SubSnpNoHgvs subSnpNoHgvs = new SubSnpNoHgvs(25928972L, 14718243L, "A", "C", ASSEMBLY, BATCH_HANDLE, BATCH_NAME,
+                                                     CHROMOSOME, CHROMOSOME_START, CONTIG_NAME, CONTIG_START,
+                                                     DbsnpVariantType.SNV, Orientation.FORWARD, Orientation.FORWARD,
+                                                     Orientation.FORWARD, false, false, false, false, CREATED_DATE,
+                                                     CREATED_DATE, TAXONOMY);
 
         DbsnpVariantsWrapper dbsnpVariantsWrapper = processor.process(subSnpNoHgvs);
-        assertSubmittedVariantDeclustered(dbsnpVariantsWrapper);
-        assertDeclusterOperationsRegistered(subSnpNoHgvs, dbsnpVariantsWrapper);
+
+        //Check Decluster
+        assertEquals(1, dbsnpVariantsWrapper.getSubmittedVariants().size());
+        assertNull(dbsnpVariantsWrapper.getSubmittedVariants().get(0).getClusteredVariantAccession());
+
+        //Check Operations
+        assertOperations(dbsnpVariantsWrapper, DECLUSTERED_ALLELES_MISMATCH, subSnpNoHgvs.getSsId(), 0);
+
+        //Check Rs
         assertEquals(subSnpNoHgvs.getRsId(), dbsnpVariantsWrapper.getClusteredVariant().getAccession());
     }
 
-    private void assertSubmittedVariantDeclustered(DbsnpVariantsWrapper dbsnpVariantsWrapper) {
-        assertEquals(1, dbsnpVariantsWrapper.getSubmittedVariants().size());
+    @Test
+    public void processSubmittedVariantTypeMismatch() throws Exception {
+        SubSnpNoHgvs subSnpNoHgvsTypeMismatch = new SubSnpNoHgvs(25928972L, 14718243L, "G", "G/-/T/TT", ASSEMBLY,
```

Comment:
Looking at the alleles, I see a SNP (G -> T), a deletion (G -> -) and an insertion (G -> TT). The dbsnp type is DIV, so both the insertion and deletion will be under that type. Why is one of those declustered? is the RS type the one of the first found (deletion or insertion) and then the other is failing?

Reply:
The dbsnpType is DIV so the EVA type of the RS is determined in `DbsnpClusteredVariantEntity` using the method `getVariantTypeWhenDiv` in this case and the result would be `DEL`. 

This is because when the dbsnpType is DIV we will determine the RS EVA type according to the type of the first variant which is INS, DEL or INDEL type. 

In this test the RS type would be DEL so the variants with type SNP and INS would be declustered for type mismatch.

Reply:
This answer addresses on of my comments too.